### ISSUE TITLE
ERROR in SecretService priority(cls)

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#122]
The availability checks in priority fail, because dbus_init() is called before the DISPLAY variable is checked for.

```
try:
  bus = secretstorage.dbus_init()
  secretstorage.Collection(bus)
except secretstorage.exceptions.SecretServiceNotAvailableException:
  raise RuntimeError("Unable to get initialize SecretService")
if 'DISPLAY' not in os.environ:
  raise RuntimeError("SecretService cannot run without a DISPLAY "
    "environment variable")
return 5
```

dbus_init() already fails with an uncaught DBusException if no DISPLAY variable is set. Pls pull the if-clause ahead. It already happens in 2.0 and I also found it in your repo's default branch.

It happens if no preference over backends is configured and when dbus is in general available, but simply no X11 display has been started.

Best, Manuel

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/122
- Originally reported by: Anonymous
- Originally created at: 2013-10-24T11:40:31.478

**Comment by jaraco** [comment]
Should not be an issue with the current bzr trunk of lp:python-secretstoage. I'm going to release 1.1 soon.

The `DISPLAY` check is also redundant and can be safely removed.

---

Original comment by: Dmitry Shachnev

**Comment by jaraco** [comment]
Thanks Dmitry for the follow upstream.

Based on your analysis, I believe the current behavior is desirable. That is, the secretstorage package should raise the NotAvailableException if DISPLAY is not set. Since it doesn't, it's a good thing that the exception flows through (so it can be reported here or upstream).

I'm going to mark this as wontfix, but please do comment if there are outstanding issues not addressed by the pending 1.1 release of secretstorage.

---

Original comment by: Jason R. Coombs
